### PR TITLE
Add API for clients to check the bots and oauth apps ids

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add API for clients to check the bots and oauth apps ids

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33158

#### Related pull requests
Webapp: https://github.com/mattermost/mattermost-webapp/pull/7850

### CODE REVIEW

File: server/utils/utils.go [hunk 1]
```diff
@@ -60,3 +61,8 @@ func LoadSession(mm *pluginapi.Client, sessionID, actingUserID string) (*model.S
 	}
 	return session, nil
 }
+
+func DumpObject(c interface{}) {
```
Comment (catalintomai) [inline server/utils/utils.go:65]: Is this one for future debugging sessions?

Comment (larkox) [reply server/utils/utils.go:65]: Yes. It was not my intention to commit this change, it is something I was using locally.

Do you want me to revert this change?

Comment (catalintomai) [reply server/utils/utils.go:65]: Might be useful for others as well (and avoids the need for everyone to constantly add and remove pre-checkin similar methods). Maybe just add a comment re: its purpose.

**Review by catalintomai — APPROVED** [review]

**Review by hanzei — APPROVED** [review]
LGTM

0/5 Maybe the API endpoints should take into account that the API might have to be enhanced in the future. Do we need a struct that contains the data instead of a list?

**Comment by larkox** [comment]
I prefer the simplest the better, but you are right that having this in a struct may make this more expandable. Worst case scenario, we can always open a new api endpoint if we need more information.

File: server/http/restapi/get_ids.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package restapi
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+func (a *restapi) handleGetBotIDs(w http.ResponseWriter, r *http.Request) {
+	apps := a.proxy.GetInstalledApps()
```
Comment (levb) [inline server/http/restapi/get_ids.go:9]: We should require user authentication on these APIs, or did I miss it?

**Review by levb — CHANGES_REQUESTED** [review]

**Review by hanzei — APPROVED** [review]

**Review by levb — APPROVED** [review]

**Pull request merged**